src/main/java/org/stellar/sdk/TransactionBuilder.java

```diff
@@ -151,11 +129,7 @@ public TransactionBuilder addTimeBounds(@NonNull TimeBounds timeBounds) {
    * @param timeout Timeout in seconds.
    * @return updated Builder
    * @see TimeBounds
-   * @deprecated this method will be removed in upcoming releases, use <code>addPreconditions()
```

Comment:
It is worth noting separately that I did not remove this function because it is one of the most commonly used functions in this class. I prefer to keep it for easier access. If you have a different opinion, please let me know.

Reply:
Understood, to maintain this, you can see the methods have to implement that unfortunate check/throw on the internal preconditions anyways which could be confusing for users, and retaining this leads to clutter in the sdk interface with multiple ways for users to set time bounds on a tx rather than a single approach. I would opt for less code and removing this and nudging sdk users to use preconditions instead since the underlying xdr was modeled to that, but defer to your preference as you noted for sdk ux.

Reply:
I agree with removing `TransactionBuilder#addTimeBounds(TimeBounds)`, but from the user's perspective, I still hope to keep `setTimeout`. I made some modifications in ca73373ae15554fbb0f4188beb0469aa2ca28bac so that we can both keep it and possibly simplify the code to make it look less complicated. If this looks good, I will merge it.

Reply:
ok, sounds good.